src/adaptors/tcp_lite/tcp_lite.c

```diff
@@ -84,6 +84,31 @@ static uint64_t buffer_threshold_50;
 static uint64_t buffer_threshold_75;
 static uint64_t buffer_threshold_85;
 
+// Window Flow Control
+//
+// This adaptor uses a simple window with acknowledge algorithm to enforce backpressure on the TCP sender. The ingress
+// adaptor will only permit up to TCP_MAX_CAPACITY bytes received before backpressuring the sender by no longer granting
+// empty receive buffers to the raw connection. The egress adapter will send its count of total bytes written out the
+// raw connection every TCP_ACK_THRESHOLD written bytes. The egress sends this update via a PN_RECEIVED frame, setting
+// the section_offset field to the total number of bytes written out the raw connection. When the PN_RECEIVED update is
+// received at the ingress adaptor it will update the window size by subtracting the section_offset from its total
+// received bytes counter. If the result of the subtraction is less than TCP_MAX_CAPACITY then backpressure is relieved
+// and empty receive buffers are given to the raw connection.
+//
+// TCP_MAX_CAPACITY: this is set to 2x the maximum number of bytes a cut through message can buffer. This makes the
+// window large enough to max out a message at each end of the TCP flow.
+//
+#define TCP_FULL_MSG      (QD_BUFFER_DEFAULT_SIZE * UCT_SLOT_COUNT * UCT_SLOT_BUF_LIMIT)
+#define TCP_MAX_CAPACITY  (TCP_FULL_MSG * UINT64_C(2))
+#define TCP_ACK_THRESHOLD TCP_FULL_MSG
```

Comment:
Can we call this TCP_ACK_THRESHOLD_BYTES since this is a byte counter ? Seems really clear that it is a BYTE counter

Reply:
What is UCT in UCT_SLOT_BUF_LIMIT ? I know UCT was already there but I just cant make out what it is by reading the name of the constant.

Reply:
"UCT" is short for "unicast cut through". It's used in all the cut through related stuff - it's better described in message.h where those symbols are defined.